src/components/expression/expression-panel.js

```diff
@@ -272,8 +285,16 @@ genotet.ExpressionPanel.prototype.formatGeneInput = function(isGeneRegex,
       });
   } else {
     var inputWords = geneInput.split(',');
-    geneNames = inputWords.filter(function(word) {
-      return word in this.data.matrixInfo.allGeneNames;
+    var lowerGeneNames = {};
+    Object.keys(this.data.matrixInfo.allGeneNames).forEach(function(name) {
```

Comment:
This isn't a good idea. You should process your lowercase dictionary only once, instead of every time user input changes.